src/client-ocurl/config.ml

```diff
@@ -1,83 +1,30 @@
-open Common_
-
 type t = {
-  debug: bool;
-  url_traces: string;
-  url_metrics: string;
-  url_logs: string;
-  headers: (string * string) list;
-  batch_timeout_ms: int;
   bg_threads: int;
+      (** Are there background threads, and how many? Default [4]. This will be
+          adjusted to be at least [1] and at most [32]. *)
   ticker_thread: bool;
+      (** If true, start a thread that regularly checks if signals should be
+          sent to the collector. Default [true] *)
   ticker_interval_ms: int;
-  self_trace: bool;
+      (** Interval for ticker thread, in milliseconds. This is only useful if
+          [ticker_thread] is [true]. This will be clamped between [2 ms] and
+          some longer interval (maximum [60s] currently). Default 500.
+          @since 0.7 *)
+  common: Client.Config.t;
+      (** Common configuration options
+          @since 0.12*)
 }
 
-let pp out self =
-  let pp_header ppf (a, b) = Format.fprintf ppf "@[%s: @,%s@]@." a b in
-  let ppheaders = Format.pp_print_list pp_header in
-  let {
-    debug;
-    url_traces;
-    url_metrics;
-    url_logs;
-    headers;
-    batch_timeout_ms;
-    bg_threads;
-    ticker_thread;
-    ticker_interval_ms;
-    self_trace;
-  } =
-    self
-  in
-  Format.fprintf out
-    "{@[ debug=%B;@ url_traces=%S;@ url_metrics=%S;@ url_logs=%S;@ \
-     headers=%a;@ batch_timeout_ms=%d; bg_threads=%d;@ ticker_thread=%B;@ \
-     ticker_interval_ms=%d;@ self_trace=%B @]}"
-    debug url_traces url_metrics url_logs ppheaders headers batch_timeout_ms
-    bg_threads ticker_thread ticker_interval_ms self_trace
+let pp fmt _ = Format.pp_print_string fmt "TODO"
```

Comment:
I'll need another pass here to complete this TODO, at minimum, but I will do another pass on the docs etc. too. But I wanted to get the code visible for input first before spending more time on the finish to ensure this general approach is acceptable.